### ISSUE TITLE
Bug in _bootstrap_xentropy_single

### ISSUE THREAD

**Issue by ismael-elatifi** [PingoLH/FCHarDNet#30]
In loss.py in function _bootstrap_xentropy_single, there is a bug in the following code portion :  
```python
        if sorted_loss[K] > thresh:  # the bug is here, it should be `sorted_loss[K] <= thresh`
            loss = sorted_loss[sorted_loss > thresh]
        else:
            loss = sorted_loss[:K]
```
Because with your current code, if the Kth loss value is below threshold it will be kept (else clause), whereas you don't want that.

**Comment by PingoLH** [comment]
Hello, thank you for the comment. I think it is just a different logic for loss extraction. Our implementation is a little bit different than the original bootstrap loss. Here is our logic:
1. extract all losses larger than the threshold
2. at least K elements for each instance of a mini-batch must be extracted because it would be biased to train the model with only a few losses.

**Comment by ismael-elatifi** [comment]
Ok so in that case you don't need the if else bloc above and you just need :  
`loss = sorted_loss[:K]`
Is it correct ?

**Comment by ismael-elatifi** [comment]
Ok I see, my bad. So indeed it is not a bug.
I close the issue which is actually not one :-)
